sixtyfps_runtime/corelib/animations.rs

```diff
@@ -68,30 +68,54 @@ impl core::ops::SubAssign<core::time::Duration> for Instant {
     }
 }
 
+#[cfg(feature = "std")]
+static INITIAL_INSTANT: once_cell::sync::OnceCell<instant::Instant> =
+    once_cell::sync::OnceCell::new();
+
 impl Instant {
     /// Returns the amount of time elapsed since an other instant.
     ///
     /// Equivalent to `self - earlier`
     pub fn duration_since(self, earlier: Instant) -> core::time::Duration {
         self - earlier
     }
+
+    /// Wrapper around [`std::time::Instant::now()`] that delegates to the backend
+    /// and allows working in no_std environments.
+    pub fn now() -> Self {
+        Self(Self::duration_since_start().as_millis() as u64)
+    }
+
+    fn duration_since_start() -> core::time::Duration {
+        #[cfg(feature = "std")]
+        {
+            let the_beginning = *INITIAL_INSTANT.get_or_init(|| instant::Instant::now());
+            return instant::Instant::now() - the_beginning;
+        }
+        #[cfg(not(feature = "std"))]
+        crate::backend::instance().unwrap().duration_since_start()
```

Comment:
why not always delegate to the backend, but the default implementation would use instand::Instant

Reply:
The main issue with that is that for the desktop builds we may not always have a backend yet. The API crate initializes it but we use timers early on in main.

Reply:
and then we don't even need to store initial_instant in this crate.

Reply:
Ok :)